### DISCUSSION TITLE
Inner Class의 특징에 대해 알려주세요.

### DISCUSSION THREAD

**Discussion by 3Juhwan** [2024-woowacourse-study/level-interview#96] [Java]
.

**Comment by gyungchan-jo** [comment]
1. 외부 클래스와의 강한 연관성: Inner 클래스는 외부 클래스의 멤버 변수와 메서드에 쉽게 접근할 수 있습니다. 

2. 코드의 구조화: Inner 클래스를 사용하면 관련된 클래스들을 논리적으로 그룹화할 수 있습니다. 이는 코드의 구조화를 향상시켜 가독성을 높이고 유지보수성을 향상시킵니다.

3. 캡슐화: Inner 클래스를 사용하면 외부 클래스의 멤버를 숨길 수 있습니다. 이는 Inner 클래스가 외부 클래스의 구현 세부 사항에 의존하지 않고 독립적으로 작동할 수 있도록 합니다.

4. Static Inner 클래스: Static Inner 클래스는 외부 클래스의 인스턴스에 종속적이지 않으며, 외부 클래스의 static 멤버에만 접근할 수 있습니다. Static Inner 클래스는 외부 클래스와 독립적으로 사용될 수 있습니다.

**Comment by le2sky** [comment]
내부 클래스는 클래스 내에 선언된 클래스입니다. 두 클래스가 긴밀한 관계에 있을 때, 사용할 수 있습니다.
내부 클래스의 장점은 다음과 같습니다.
한 클래스를 다른 클래스의 내부 클래스로 선언하면 두 클래스의 멤버들 간에 서로 쉽게 접근할 수 있습니다.
외부에 불필요한 클래스를 감춰 코드의 복잡성을 줄일 수 있습니다.

내부 클래스의 종류는 크게 4가지가 있습니다.

instance class : outer 클래스의 멤버변수 선언위치에 선언하며, outer 클래스의 인스턴스 멤버처럼 다뤄집니다. 
static class : outer 클래스의 멤버변수 선언위치에 선언하며, outer 클래스의 static 멤버처럼 다뤄집니다.
local class : outer 클래스의 메서드나 초기화블럭 안에서 선언하며, 선언된 영역 내부에서만 사용될 수 있습니다.
anonymous class : 클래스의 선언과 객체의 생성을 동시에 하는 이름없는 클래스입니다.

내부 클래스의 선언위치에 따라서 같은 선언위치의 변수와 동일한 스코프와 접근성을 가집니다.
내부 클래스 중에서 유일하게 static 클래스만 static 멤버를 가질 수 있습니다. (final static은 모든 내부 클래스에서 ok!!!!)

**Comment by GIVEN53** [comment]
클래스간 연관성 높을 때 그룹화
멤버변수 접근 쉬움
캡슐화
